src/TextArea/TextArea.jsx

```diff
@@ -0,0 +1,185 @@
+            htmlFor={id}
+            variant='inputLabel'
+            width='fit-content'>
+            {label} {required ? '*' : ''}
+          </Typography>
+        ) : null}
+        <Flex alignItems='center' position='relative' width='100%'>
+          <StyledTextArea
+            autoComplete={autoComplete}
+            autoFocus={autoFocus}
+            disabled={readOnly || disabled}
+            error={error}
+            hasLabel={label}
+            hasIcLeft={icLeft}
+            hasIcRight={icRight}
+            id={id}
+            name={name}
+            onChange={onChange}
+            placeholder={readOnly ? '' : placeholder}
+            readOnly={readOnly}
+            ref={ref || textAreaRef}
+            required={required}
+            type={type}
+            value={value}
+            variant={textAreaVariant}
+            allowResize={allowResize}
+            maxLength={maxLength}
+            cols={cols}
+            rows={rows}
+            {...rest}
+          />
+        </Flex>
+        <ErrorMessage error={error} preserveSpace={!noBottomSpace} />
+      </Flex>
+    )
+  }
+)
+
+const StyledTextArea = styled.textarea(
+  css({
+    ...typography.labelSmall,
+    display: 'block',
+    width: '100%',
+    borderWidth: 1,
+    borderStyle: 'solid',
+    borderColor: 'outline',
+    borderRadius: 8,
+    outline: 'none',
+    color: 'text',
+    padding: 3,
+    ':hover, :focus': {
+      borderWidth: 1,
+      borderStyle: 'solid'
+    }
+  }),
+  ({ allowResize }) =>
+    !allowResize &&
+    css({
+      resize: 'none'
+    }),
+  ({ error }) =>
+    error
+      ? css({
+          color: 'error',
+          borderColor: 'error'
+        })
+      : null,
+  ({ disabled }) =>
+    disabled
+      ? css({
+          opacity: 0.5
+        })
+      : null
+)
+
+TextArea.propTypes = {
+  ...propTypes.inputStyle,
+  allowResize: PropTypes.bool,
+  maxLength: PropTypes.number,
+  cols: PropTypes.number,
+  rows: PropTypes.number,
+  autoComplete: PropTypes.string,
+  autoFocus: PropTypes.bool,
+  disabled: PropTypes.bool,
+  error: PropTypes.oneOfType([
+    PropTypes.bool,
+    PropTypes.node,
+    PropTypes.string
+  ]),
+  icLeft: PropTypes.string,
+  icRight: PropTypes.string,
+  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+  label: PropTypes.node,
+  name: PropTypes.string,
+  /** dummy space added for consistent spacing with validated inputs.
+   *
+   * remove space by setting this to true */
+  noBottomSpace: PropTypes.bool,
+  onChange: PropTypes.func,
+  placeholder: PropTypes.string,
+  placeholderTextColor: PropTypes.string,
+  readOnly: PropTypes.bool,
+  required: PropTypes.bool,
+  type: PropTypes.string,
+  valid: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
+  value: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number,
+    PropTypes.object
+  ])
+}
+
+TextArea.defaultProps = {
+  autoFocus: false,
+  rows: 5,
+  allowResize: false,
```

Comment:
Default HTML behaviour is to allow resize.

```suggestion
  allowResize: true,
```